docs/release/trg-0/trg-1-4.md

```diff
@@ -0,0 +1,94 @@
+---
+title: TRG 1.04 - Diagrams as Code
+---
+
+| Status | Created       | Post-History          |
+|--------|---------------|-----------------------|
+| Draft  | 20-Sept-2023  | created initial Draft |
+|        |               |                       |
+
+## Why
+
+To have an easier approach to update current static files and have a better overview of the changes within the product pictures.
+
+This helps people to understand our workflows within our products and make it easier to understand the product itself.
+Also updating the static files will be easier and faster to do for every committer or maintainer of our repositories.
+
+## Description
+
+All static files **must** be created with code and also stored as `.svg` file within your product repository.
+After a committers [decision](https://github.com/eclipse-tractusx/sig-infra/discussions/19) there are 2 languages accepted, either [PUML](https://plantuml.com/en/) or [Mermaid](https://mermaid.js.org/).
+These code files (`.puml`,`.mmd`,`mermaid`) and static `.svg` files **must** be stored within the `docs/static` folder.
```

Comment:
Thanks for appreciation.

I also agree that is bit harsh, but also refered to our TRG 2.03 `/docs` folder description so would remove the `static` folder. Additionally to the point with `all` is a bit confusing, your right.

to side note: already adpted the syntax from markdown without blank in the code reference. 😉 

edited a bit that there are more options how to store, but like the idea to have all diagrams as a 'must' so that we are all aligend on the diagram tooling. 

https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/401/commits/160bbdf305dc6b3bc0b7d07b1821894ee116634f

Reply:
Thanks for your response and excluding the 'static' folder!

Unfortunately, I'm still not on board with the **must** character for such a TRG: IMO we should establish 'must'-TRGs only if really necessary. 
As mentioned above, we're risking to alienate potential contributors and this TRG doesn't provide enough benefit to outweigh that risk.
IMO this TRG should have a guideline character not a 'must' character.

Reply:
Kay,  i can understand your feeling about new members and to much strict `guidelines` within our TRG's.

Personally i like the idea to have a clear way where to go and how to store, espacially with all our how to call wild grewed things about documentation and storing fancy diagrams everywhere. 😉 

But will try to get another feedback round for how strict the diagrams as code attitude have to be.

Reply:
@evegufy is your issue with the static folder or with the selected technologies?

I don't think it has to be in a specific folder aslong as the diagrams used are changeable by anyone.

The must character for technologie is something we want to have to align the documentation across tractus-x. The origin of this trg came from a meeting with business.

Reply:
I also like the outcome of not defining the docs/static folder - in case you're e.g. documenting with arc42, you may want to store files according to your document structure in the respective section

Reply:
@tom-rm-meyer-ISST Thanks for feedback, was already changed to `/docs` folder as stated in https://eclipse-tractusx.github.io/docs/release/trg-2/trg-2-3#docs
https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/401/commits/160bbdf305dc6b3bc0b7d07b1821894ee116634f

Reply:
@evegufy sry for my late reply: This trg is an outcome between a few people including Daniel.

I do believe its part of a Quality Aspect (and the representation of it) that the documentation doesn't look like its written from a 1000 different people not knowing each other. Diagrams are very prominent and visual.

Often enough people don't care for a specific tool but just use one randomly.

But there are hard requirements from an Open Source perspective: If you can't edit it, you can't fix a bug/issue with the diagram and you need to recreate it.  For this alone the requirement "Has to be done with an open source tool" and "the original file has to be available" are a must not a should.